pkg/virt-api/rest/objectgraph.go

```diff
@@ -0,0 +1,229 @@
+		Items: resources,
+	}
+
+	if len(errs) > 0 {
+		return objectGraph, k8serrors.NewAggregate(errs)
+	}
+	return objectGraph, nil
+}
+
+func (og *ObjectGraph) virtualMachineInstanceBackupGraph(vmi *v1.VirtualMachineInstance) (v1.ObjectGraphNodeList, error) {
+	var resources []v1.ObjectGraphNode
+	var errs []error
+
+	resources, err := og.addLauncherPod(vmi.GetName(), vmi.GetNamespace(), resources)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	resources = og.addAccessCredentials(vmi.Spec.AccessCredentials, vmi.GetNamespace(), resources)
+	resources, err = og.addVolumeGraph(vmi, vmi.GetNamespace(), resources)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	objectGraph := v1.ObjectGraphNodeList{
+		Items: resources,
+	}
+
+	if len(errs) > 0 {
+		return objectGraph, k8serrors.NewAggregate(errs)
+	}
+	return objectGraph, nil
+}
+
+func (og *ObjectGraph) addVolumeGraph(obj interface{}, namespace string, resources []v1.ObjectGraphNode) ([]v1.ObjectGraphNode, error) {
+	volumes, err := storageutils.GetVolumes(obj, og.client, storageutils.WithAllVolumes)
+	if err != nil {
+		return resources, err
+	}
+
+	for _, volume := range volumes {
+		switch {
+		case volume.DataVolume != nil:
+			resources = og.addGraphNode(volume.DataVolume.Name, namespace, "datavolumes", resources)
+			resources = og.addGraphNode(volume.DataVolume.Name, namespace, "persistentvolumeclaims", resources)
+		case volume.PersistentVolumeClaim != nil:
+			resources = og.addGraphNode(volume.PersistentVolumeClaim.ClaimName, namespace, "persistentvolumeclaims", resources)
+		case volume.MemoryDump != nil:
+			resources = og.addGraphNode(volume.MemoryDump.ClaimName, namespace, "persistentvolumeclaims", resources)
+		case volume.ConfigMap != nil:
+			resources = og.addGraphNode(volume.ConfigMap.Name, namespace, "configmaps", resources)
+		case volume.Secret != nil:
+			resources = og.addGraphNode(volume.Secret.SecretName, namespace, "secrets", resources)
+		case volume.ServiceAccount != nil:
+			resources = og.addGraphNode(volume.ServiceAccount.ServiceAccountName, namespace, "serviceaccounts", resources)
+		}
+	}
+	return resources, err
+}
+
+func (og *ObjectGraph) addAccessCredentials(acs []v1.AccessCredential, namespace string, resources []v1.ObjectGraphNode) []v1.ObjectGraphNode {
+	for _, ac := range acs {
+		if ac.SSHPublicKey != nil && ac.SSHPublicKey.Source.Secret != nil {
+			resources = og.addGraphNode(ac.SSHPublicKey.Source.Secret.SecretName, namespace, "secrets", resources)
+		} else if ac.UserPassword != nil && ac.UserPassword.Source.Secret != nil {
+			resources = og.addGraphNode(ac.UserPassword.Source.Secret.SecretName, namespace, "secrets", resources)
+		}
+	}
+	return resources
+}
+
+func (og *ObjectGraph) addLauncherPod(vmiName, vmiNamespace string, resources []v1.ObjectGraphNode) ([]v1.ObjectGraphNode, error) {
+	pods, err := og.client.CoreV1().Pods(vmiNamespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "kubevirt.io=virt-launcher",
```

Comment:
Maybe also add `kubevirt.io/vm: <name of vmi>` as a label selector? That should cut down on the number of pods you are grabbing.